#!/usr/bin/env python
PACKAGE = "psaf_local_planner"

from dynamic_reconfigure.parameter_generator_catkin import *
 
gen = ParameterGenerator()

gen.add("closest_point_local_plan", double_t, 0, "How far the clostest point is away from the car, ensures the local plan is always ahead of the car", 2.,  0.5, 5.)
gen.add("lookahead_factor", double_t, 0, "How far the car should look forward; lower is further", 3,  0.1, 10.)
gen.add("min_velocity", double_t, 0, "Lowest speed the car should be able to go, mainly for curves", 5.,  1., 20.)
gen.add("estimate_curvature_distance", double_t, 0, "Distance forward in which the curvature should be estiamted", 30., 10., 50.)
gen.add("check_collision_max_distance", double_t, 0, "Max Distance forward it should check for collsisions", 40.,  10., 60.)

exit(gen.generate(PACKAGE, "psaf_local_planner", "PsafLocalPlannerParameter"))